cdap-common/bin/functions.sh

```diff
@@ -503,15 +503,31 @@ cdap_set_spark() {
       # If there is no valid SPARK_HOME, we should unset the existing one if it is set
       # Otherwise the spark-shell won't run correctly
       unset SPARK_HOME
+      local __spark_shell=$(which spark-shell)
+      local __spark_client_version=None
+      for __dist in hdp iop; do
```

Comment:
The default for `__spark_shell` is the output from `which spark-shell` (though I do need to redirect stderr away... I'll fix)... on CDH, it never matches. However, there's a bug near here. I'll fix that, also.